File: code_samples/typescript/error_handling_custom_exceptions.ts
/**
 * Error Handling: Custom Exceptions
 * AI/ML Training Sample
 */

interface ICustomExceptions {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class CustomExceptions implements ICustomExceptions {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new CustomExceptions();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { CustomExceptions, ICustomExceptions };

File: code_samples/elixir/error_handling_custom_exceptions.ex
# Error Handling: Custom Exceptions
# AI/ML Training Sample

defmodule CustomExceptions do
  defstruct data: ""
  
  def new(), do: %CustomExceptions{}
  
  def process(%CustomExceptions{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%CustomExceptions{data: data}), do: data
  
  def validate(%CustomExceptions{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = CustomExceptions.new()
updated = CustomExceptions.process(instance, "example")
IO.puts("Data: " <> CustomExceptions.get_data(updated))
IO.puts("Valid: " <> to_string(CustomExceptions.validate(updated)))
